src/main/java/org/folio/rest/impl/TenantApiImpl.java

```diff
@@ -0,0 +1,80 @@
+package org.folio.rest.impl;
+
+import io.vertx.core.AsyncResult;
+import io.vertx.core.Context;
+import io.vertx.core.Future;
+import io.vertx.core.Handler;
+import io.vertx.core.logging.Logger;
+import io.vertx.core.logging.LoggerFactory;
+import org.apache.commons.io.IOUtils;
+import org.apache.commons.lang.StringUtils;
+import org.folio.rest.annotations.Validate;
+import org.folio.rest.jaxrs.model.TenantAttributes;
+import org.folio.rest.persist.PostgresClient;
+import org.folio.rest.tools.utils.TenantTool;
+import javax.ws.rs.core.Response;
+import java.io.IOException;
+import java.io.InputStream;
+import java.nio.charset.StandardCharsets;
+import java.util.List;
+import java.util.Map;
+
+public class TenantApiImpl extends TenantAPI {
+
+  private static final String TEST_DATA_SQL = "templates/db_scripts/test-data.sql";
+  private static final String OKAPI_TENANT_HEADER = "x-okapi-tenant";
+  private static final String TENANT_PLACEHOLDER = "${myuniversity}";
+  private static final String MODULE_PLACEHOLDER = "${mymodule}";
+  private static final String TEST_MODE = "test.mode";
+  private final Logger logger = LoggerFactory.getLogger(TenantApiImpl.class);
+
+  @Validate
+  @Override
+  public void postTenant(TenantAttributes entity, Map<String, String> headers, Handler<AsyncResult<Response>> handlers,
+                         Context context) {
+    super.postTenant(entity, headers, result -> {
+      if (result.failed()) {
+        handlers.handle(result);
+      } else {
+        setupTestData(headers, context).setHandler(event -> handlers.handle(result));
+      }
+    }, context);
+  }
+
+  private Future<List<String>> setupTestData(Map<String, String> headers, Context context) {
+    try {
+
+      if (!Boolean.TRUE.equals(Boolean.valueOf(System.getenv(TEST_MODE)))) {
```

Comment:
Just curious -- is this a commonly used approach for controlling whether test data is loaded or not?

Reply:
it is a custom approach

Reply:
Ok, thanks -- so by default test data will not get loaded. Environment variable is needed for this as you noted in the sample deployment descriptor from previous PR -- got it